sql/src/test/java/org/apache/druid/sql/calcite/CalciteArraysQueryTest.java

```diff
@@ -4807,4 +4808,187 @@ public void testUnnestWithGroupByHavingWithWhereOnUnnestCol()
+                                    .context(QUERY_CONTEXT_UNNEST)
+                                    .build()
+                          ),
+                          new QueryDataSource(
+                              Druids.newScanQueryBuilder()
+                                    .dataSource(new TableDataSource(CalciteTests.DATASOURCE1))
+                                    .intervals(querySegmentSpec(Filtration.eternity()))
+                                    .resultFormat(ScanQuery.ResultFormat.RESULT_FORMAT_COMPACTED_LIST)
+                                    .legacy(false)
+                                    .context(QUERY_CONTEXT_UNNEST)
+                                    .columns("__time", "cnt", "dim1", "dim2", "dim3", "m1", "m2", "unique_dim1")
+                                    .build()
+                          ),
+                          "_j0.",
+                          "1",
+                          JoinType.INNER,
+                          null
+                      )
+                  )
+                  .intervals(querySegmentSpec(Filtration.eternity()))
+                  .resultFormat(ScanQuery.ResultFormat.RESULT_FORMAT_COMPACTED_LIST)
+                  .legacy(false)
+                  .columns("j0.unnest")
+                  .context(QUERY_CONTEXT_UNNEST)
+                  .build()
+        ),
+        NullHandling.sqlCompatible() ?
+        ImmutableList.of(
+            new Object[]{"a"},
+            new Object[]{"a"},
+            new Object[]{"a"},
+            new Object[]{"a"},
+            new Object[]{"a"},
+            new Object[]{"a"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"b"},
+            new Object[]{"c"},
+            new Object[]{"c"},
+            new Object[]{"c"},
+            new Object[]{"c"},
+            new Object[]{"c"},
+            new Object[]{"c"},
+            new Object[]{"d"},
+            new Object[]{"d"},
+            new Object[]{"d"},
+            new Object[]{"d"},
+            new Object[]{"d"},
+            new Object[]{"d"},
+            new Object[]{""},
+            new Object[]{""},
+            new Object[]{""},
+            new Object[]{""},
+            new Object[]{""},
+            new Object[]{""},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null},
+            new Object[]{null}
```

Comment:
Can reconcile the two branches by instead using `NullHandling.defaultStringValue()` Would that help reconcile the two branches? 
```suggestion
            new Object[]{NullHandling.defaultStringValue()},
....
```